akka-cluster-sharding/src/main/scala/akka/cluster/sharding/Shard.scala

```diff
@@ -962,13 +962,17 @@ private[akka] class Shard(
                     entities.pendingRememberEntities())
                 appendToMessageBuffer(entityId, msg, snd)
                 entities.rememberingStart(entityId, ackTo = None)
-              } else {
+              } else if (rememberEntitiesStore.isDefined) {
                 // No actor running and no write in progress, start actor and deliver message when started
                 if (verboseDebug)
                   log.debug("Buffering message [{}] to [{}] and starting actor", payload.getClass, entityId)
                 appendToMessageBuffer(entityId, msg, snd)
                 entities.rememberingStart(entityId, ackTo = None)
                 rememberUpdate(add = Set(entityId))
+              } else {
+                // don't buffer if remember entities not enabled
+                getOrCreateEntity(entityId).tell(payload, snd)
```

Comment:
To avoid an add to buffer then immediately unbuffering. 

These are recorded in cinnamon so shows up for every new entity when not using remembered entities

Reply:
Perhaps put the non-RE-case up top instead, since that should A be more common and perhaps make it easier to jit the other branches away (maybe we should even encode `isRememberingEntities` as a boolean field from the constructor)?

Reply:
👍